solutions/usaco-1063.mdx

```diff
@@ -16,63 +16,146 @@ author: Kevin Sheng
 Note: the algorithm in the code may differ from that described in the video.
 
 <LanguageSection>
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <cassert>
+#include <vector>
+#include <set>
+#include <map>
+#include <algorithm>
+
+using namespace std;
+
+int main() {
+	int cow_num;
+	std::cin >> cow_num;
+
+	set<int> seen_x, seen_y;
+	vector<pair<int, int>> cows(cow_num);
+	for (pair<int, int>& c : cows) {
+		std::cin >> c.first >> c.second;
+		assert(!(seen_x.count(c.first) || seen_y.count(c.second)));
```

Comment:
whys this assert here

Reply:
input validation

Reply:
you see the good thing about cp is that it saves you this piece of mind

Reply:
it doesn't really matter, the solution still runs in time